ImplicitlyExpandDesignTimeFacades target should only add facades if a higher version isn't already referenced

The `ImplicitlyExpandDesignTimeFacades` target adds references (specifically `ReferencePath` items) to the System.Runtime surface facades under the Facades folder in the reference assembly folder.  However, we now also ship facades as part of NuGet packages.  The facades from a NuGet package may conflict with the facades that are implicitly added via this target.

An example of an issue due to this is in dotnet/sdk#367.  To repro this issue, create an ASP.NET Core web application targeting .NET Framework, update the `TargetFramework` property to `net461`, and run `dotnet publish`.  You'll get an error because a `Copy` task is trying to copy two different versions of `System.Runtime.Serialization.Primitives.dll` to the publish folder.

I discussed this with @weshaggard and it is likely that in the future we will ship updated facades in the targeting packs that should be chosen over previous versions in NuGet packages.  So we likely need a task that examines the `ReferencePath` items that come from `ResolveAssemblyReferences` and if there are any references with the same filename as an item in the Facades folder, pick the one with the higher Assembly Version.

@cdmihai I'm trying to work around this issue. I'm publishing for `net462` using the .NET Core CLI (so `dotnet publish`).

Setting `$(ImplicitlyExpandDesignTimeFacades)` to `false` seems to cause a lot of side issues.

How would I go about and remove the duplicate assembly from `@(ReferencePath)`? I tried a naive `<ReferencePath Remove='{full path to facade dll in NuGet package}'/>" but that doesn't seem to work. Do I need to plug this in at a specific target?

Here's how my publish task is failing:

```
                   Task "Copy" (TaskId:351)
                     Task Parameter:
                         SourceFiles=
                             C:\Users\Frederik Carlier\.nuget\packages\system.diagnostics.tracing\4.3.0\ref\net462\System.Diagnostics.Tracing.dll
                                     CopyLocal=false
                                     CopyToPublishDirectory=Always
                                     FusionName=System.Diagnostics.Tracing, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
                                     ImageRuntime=v4.0.30319
                                     NuGetIsFrameworkReference=false
                                     NuGetSourceType=Package
                                     OriginalItemSpec=C:\Users\Frederik Carlier\.nuget\packages\system.diagnostics.tracing\4.3.0\ref\net462\System.Diagnostics.Tracing.dll
                                     Private=false
                                     ReferenceSourceTarget=ResolveAssemblyReference
                                     RelativePath=refs\System.Diagnostics.Tracing.dll
                                     ResolvedFrom={RawFileName}
                                     Version=4.1.1.0
                             C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6.2\Facades\System.Diagnostics.Tracing.dll
                                     CopyLocal=false
                                     CopyToPublishDirectory=Always
                                     RelativePath=refs\System.Diagnostics.Tracing.dll
                                     ResolvedFrom=ImplicitlyExpandDesignTimeFacades
                                     WinMDFile=false (TaskId:351)
                     Task Parameter:DestinationFiles=bin\Release\net462\win7-x64\publish\refs\System.Diagnostics.Tracing.dll (TaskId:351)
                     Task Parameter:OverwriteReadOnlyFiles=False (TaskId:351)
00:58:44.244     1>C:\Program Files\dotnet\sdk\1.0.0-preview5-004275\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Publish.targets(128,5): error MSB3094: "DestinationFiles" refers to 1 item(s), and "SourceFiles" refers to 2 item(s). They must have the same number of items. [C:\Users\Frederik Carlier\Source\Repos\Quamotion\src\Quamotion.WebDriver\Quamotion.WebDriver.Core.csproj]
                     Output Item(s): FileWrites=bin\Release\net462\win7-x64\publish\refs\System.Diagnostics.Tracing.dll (TaskId:351)
                   Done executing task "Copy" -- FAILED. (TaskId:351)
```

--- a/src/XMakeTasks/Microsoft.NETFramework.CurrentVersion.targets
+++ b/src/XMakeTasks/Microsoft.NETFramework.CurrentVersion.targets
@@ -108,7 +108,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
       <_DesignTimeFacadeAssemblies_Names Remove="@(_ReferencePath_Names)"/>
 
-      <ReferencePath Include="@(_DesignTimeFacadeAssemblies_Names->%(OriginalIdentity))">
+      <ReferencePath Include="@(_DesignTimeFacadeAssemblies_Names->'%(OriginalIdentity)')">
         <WinMDFile>false</WinMDFile>
         <CopyLocal>false</CopyLocal>
         <ResolvedFrom>ImplicitlyExpandDesignTimeFacades</ResolvedFrom>

--- a/src/XMakeTasks/Microsoft.NETFramework.CurrentVersion.targets
+++ b/src/XMakeTasks/Microsoft.NETFramework.CurrentVersion.targets
@@ -97,9 +97,18 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
     <ItemGroup Condition="'$(_HasReferenceToSystemRuntime)' == 'true'">
       <_DesignTimeFacadeAssemblies Include="%(DesignTimeFacadeDirectories.Identity)*.dll"/>
-      <!-- If the user has manually referenced these assemblies, remove them so we don't end up with duplicates -->
-      <ReferencePath Remove="@(_DesignTimeFacadeAssemblies)"/>
-      <ReferencePath Include="%(_DesignTimeFacadeAssemblies.Identity)">
+
+      <_DesignTimeFacadeAssemblies_Names Include="@(_DesignTimeFacadeAssemblies->'%(FileName)')">
+          <OriginalIdentity>%(_DesignTimeFacadeAssemblies.Identity)</OriginalIdentity>
+      </_DesignTimeFacadeAssemblies_Names>
+
+      <_ReferencePath_Names Include="@(ReferencePath->'%(FileName)')">
+          <OriginalIdentity>%(ReferencePath.Identity)</OriginalIdentity>
+      </_ReferencePath_Names>
+
+      <_DesignTimeFacadeAssemblies_Names Remove="@(_ReferencePath_Names)"/>
+
+      <ReferencePath Include="@(_DesignTimeFacadeAssemblies_Names->%(OriginalIdentity))">
         <WinMDFile>false</WinMDFile>
         <CopyLocal>false</CopyLocal>
         <ResolvedFrom>ImplicitlyExpandDesignTimeFacades</ResolvedFrom>

